src/main/scala/org/apache/spark/streamdm/classifiers/trees/SplitCriterion.scala

```diff
@@ -101,7 +101,9 @@ class InfoGainSplitCriterion extends SplitCriterion with Serializable {
    */
   def entropy(pre: Array[Double]): Double = {
     if (pre == null || pre.sum <= 0 || hasNegative(pre)) 0.0
-    log2(pre.sum) - pre.filter(_ > 0).map(x => x * log2(x)).sum / pre.sum
+    else {
```

Comment:
Add two braces to make code easy understand.
if (pre == nil || rep.sum <= 0 || hasNegative(pre)) {
0.0
} else {
log2(pre.sum) - pre.filter(_ > 0).map(x >= x * log2(x)).sum / pre.sum
}

Reply:
I added two brackets as suggested.